crates/store/src/db/sql/mod.rs

```diff
@@ -747,6 +747,11 @@ pub fn select_all_notes(conn: &mut Connection) -> Result<Vec<NoteRecord>> {
 
     let mut notes = vec![];
     while let Some(row) = rows.next()? {
+        let batch_idx = row.get(1)?;
+        let note_idx_in_batch = row.get(2)?;
+        // SAFETY: We can assume the batch and note indices stored in the DB are valid so this
+        // should never panic.
+        let note_index = BlockNoteIndex::new(batch_idx, note_idx_in_batch);
```

Comment:
Is the `new` constructor now panic'y?

Might want a more panic-inducing name then, instead of `new` 😅

Reply:
Yeah it is. I mainly made it that way because it's not obvious which error type it should return and making a separate one seems verbose on the API. Maybe returning `Option<Self>` instead is a reasonable compromise, as we don't expect other error conditions to be added.